src/lib/storage/dictionary_column.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "base_column.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+class BaseAttributeVector;
+
+// Even though ValueIDs do not have to use the full width of ValueID (uint32_t), this will also work for smaller ValueID
+// types (uint8_t, uint16_t) since after a down-cast INVALID_VALUE_ID will look like their numeric_limit::max()
+constexpr ValueID INVALID_VALUE_ID{std::numeric_limits<ValueID::base_type>::max()};
+
+// Dictionary is a specific column type that stores all its values in a vector
+template <typename T>
+class DictionaryColumn : public BaseColumn {
+ public:
+  /**
+   * Creates a Dictionary column from a given value column.
+   */
+  explicit DictionaryColumn(const std::shared_ptr<BaseColumn>& base_column);
+
+  // SEMINAR INFORMATION: Since most of these methods depend on the template parameter, you will have to implement
+  // the DictionaryColumn in this file. Replace the method signatures with actual implementations.
```

Comment:
True, we overlooked this. We thought we were supposed to use the `EXPLICITLY_INSTANTIATE_COLUMN_TYPES` macro again, since we are dealing with columns here as well. @TeachingAssistants what is the reason why we were not supposed to do this?